test/Microsoft.TemplateEngine.TestTemplates/test_templates/PostActions/WithFileRename/.template.config/template.json

```diff
@@ -0,0 +1,34 @@
+{
+  "$schema": "https://json.schemastore.org/template.json",
+  "author": "Test Asset",
+  "classifications": [ "Test Asset" ],
+  "name": "TestAssets.PostActions.AddJsonProperty.WithSourceNameChangeInJson",
+  "generatorVersions": "[1.0.0.0-*)",
+  "groupIdentity": "TestAssets.PostActions.AddJsonProperty.WithSourceNameChangeInJson",
+  "precedence": "100",
+  "identity": "TestAssets.PostActions.AddJsonProperty.WithSourceNameChangeInJson",
+  "shortName": "TestAssets.PostActions.AddJsonProperty.WithSourceNameChangeInJson",
+  "tags": { "type": "project" },
+  "sourceName": "MyTestProject",
+  "primaryOutputs": [
+    {
+      "path": "MyTestProject.csproj"
+    }
+  ],
+  "postActions": [
+    {
+      "description": "Modify JSON file",
+      "manualInstructions": [ { "text": "Add MyTestProject property to testfile.json manually." } ],
+      "args": {
+        "jsonFileName": "testfile.json",
+        "parentPropertyPath": "moduleConfiguration:edgeAgent:properties.desired:modules",
+        "newJsonPropertyName": "MyTestProject",
+        "newJsonPropertyValue": "${MODULEDIR<../MyTestProject>}"
+      },
+      "applyFileRenamesToArgs": [ "newJsonPropertyName", "newJsonPropertyValue" ],
```

Comment:
I wonder if `applyFileRenamesToArgs` is not a bit confusing.  You're actually not renaming any files ?
Maybe `applySourceRenamesToArgs` is better ? (Or even simply plain `applyRenamesToArgs`? )

Reply:
i used `fileRename`, as the configuration for symbol that is used is called `fileRename`, please check another example here: https://github.com/vlada-shubina/templating/blob/4c2bab980aa1f93ed00186efa6376780b08fb6b8/test/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests/TemplateConfigTests/PostActionTests.cs#L272-L307. 

`sourceName` is a special symbol, for which both `replaces` and `fileRename` value is same and corresponds to value of `sourceName`. 

We can definitely shorten it to `applyRenamesToArgs`. I would not recommend using `applySourceRenamesToArg` which may imply using [source file renames](https://github.com/dotnet/templating/wiki/Reference-for-template.json#source-definition), which are not applied in this scenario. Only symbol based renames are applied.

Reply:
Well, it was just a suggestion as the first time I saw the name, I thought that filenames were going to be renamed. :)